### PR TITLE
Pagination: Fixing paginate for /api/v1/superuser/logs API

### PR DESCRIPTION
**Pull request by Sunandadadi**
Superuser logs are sorted on datetime field. The next page token contains the value of the start_index of the datetime field for the next page. datetime column on the LogEntry table is not unique, so in some cases,  multiple entries can have the same datetime field. 
For example, logs have a page limit of 20 entries. If 25 entries have the same datetime - d1, the start_index of the next page is again d1. With the current logic, the same query is run for the next page too. This PR addresses such issues and adds an offset in such cases where (the first 20 entries of page 1 and 1 or more entries of page 2 have the same datetime).

### CODE REVIEW

File: data/model/modelutil.py [hunk 1]
```diff
@@ -69,24 +77,33 @@ def pagination_start(page_token=None):
...
+                results[limit - 1], sort_field_name
+            )
+            offset_val = limit + offset_val if use_offset else 0
             is_datetime = True
 
         page_token = {
             "start_index": start_index,
             "page_number": page_number + 1 if page_number else 1,
             "is_datetime": is_datetime,
+            "use_offset": use_offset,
```
Comment (kleesc) [inline data/model/modelutil.py:105]: Do you really need `use_offset`, or just to set `offset_val`?

Comment (Sunandadadi) [reply data/model/modelutil.py]: nice, I can just use `offset_val `

File: data/model/modelutil.py [hunk 2]
```diff
@@ -69,24 +77,32 @@ def pagination_start(page_token=None):
...
 
-def paginate_query(query, limit=50, sort_field_name=None, page_number=None):
+def paginate_query(query, limit=50, sort_field_name=None, page_number=None, offset_val=0):
     """
     Executes the given query and returns a page's worth of results, as well as the page token for
     the next page (if any).
     """
     results = list(query)
     page_token = None
+    offset_val = 0
```
Comment (kleesc) [inline data/model/modelutil.py:87]: This is already defined as a default parameter

Comment (Sunandadadi) [reply data/model/modelutil.py]: Ah, removed.

**Review by kleesc — APPROVED** [review]
LGTM

**Pull request merged**